internal/handlers/updates_handler.go

```diff
@@ -0,0 +1,77 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+
+	"github.com/andreevym/metric-collector/internal/logger"
+	"github.com/andreevym/metric-collector/internal/storage"
+	"go.uber.org/zap"
+)
+
+const (
+	UpdatesMetricContentType = "application/json"
+)
+
+func (s ServiceHandlers) PostUpdatesHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", UpdatesMetricContentType)
+
+	bytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		logger.Log.Error("error", zap.Error(err))
+		return
+	}
+
+	var metrics []*storage.Metric
+	err = json.Unmarshal(bytes, &metrics)
+	if err != nil {
+		logger.Log.Error("err", zap.Error(err))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if len(metrics) == 0 {
+		return
+	}
+
+	result := map[string]*storage.Metric{}
+	for _, metric := range metrics {
+		found, ok := result[metric.ID+metric.MType]
+		if ok && found != nil && metric.MType == storage.MTypeCounter {
+			newDelta := *metric.Delta + *found.Delta
+			metric.Delta = &newDelta
+		}
+
+		result[metric.ID+metric.MType] = metric
+	}
+
+	metricsR := map[string]*storage.MetricR{}
```

Comment:
лучше подобный код с большим количеством логики выносить из ручек в отдельные сервисы, тем более что можно переиспользовать
Второе - в данном случае нет смысла делать ссылки, это может привести только к проблемам (например удалена или изменена структура на которую есть ссылка), а профита не даст. А ниже из-за ссылки приходится городить плохочитаемый код
```suggestion
	metricsR := map[string]storage.MetricR{}
```

Reply:
вынес логику сохранения в storage пакет, пока что лучше ничего не придумал а еще одни абстракции городить не хочется.
код с указателями исправил.